samza-core/src/main/java/org/apache/samza/task/TaskCallbackManager.java

```diff
@@ -46,22 +49,22 @@
      * Adding the newly complete callback to the callback queue
      * Move the queue to the last contiguous callback to commit offset
      * @param cb new callback completed
-     * @return callback of highest watermark needed to be committed
+     * @return list of callbacks to be committed
      */
-    TaskCallbackImpl update(TaskCallbackImpl cb) {
+    List<TaskCallbackImpl> update(TaskCallbackImpl cb) {
+
       synchronized (lock) {
         callbacks.add(cb);
-
         TaskCallbackImpl callback = null;
-        TaskCallbackImpl callbackToCommit = null;
+        List<TaskCallbackImpl> committableCallbacks = new ArrayList<>();
         TaskCoordinator.RequestScope shutdownRequest = null;
         // look for the last contiguous callback
         while (!callbacks.isEmpty() && callbacks.peek().matchSeqNum(nextSeqNum)) {
           ++nextSeqNum;
           callback = callbacks.poll();
 
```

Comment:
Yes, this is fixed. 
I'd misunderstanding that task.commit is for commit in only the callback  SSP that is processed & committed by the task. Thanks for pointing it out that it's for all the input SSP's that are processed until the callback SSP that's processed & committed.